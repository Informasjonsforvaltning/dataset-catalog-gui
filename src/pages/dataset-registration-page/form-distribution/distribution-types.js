import { licenseType, textType } from '../../../schemaTypes';

export const distributionTypes = values => {
  let distributions = null;
  if (values && values.length > 0) {
    distributions = values.map(item => ({
      id: item.id ? item.id : '',
      description: item.description ? item.description : textType,
      accessURL: item.accessURL ? item.accessURL : [],
      license: item.license ? item.license : licenseType,
      conformsTo: item.conformsTo ? item.conformsTo : [],
      page: item.page && item.page.length > 0 ? item.page : [licenseType],
      format: item.format ? item.format : [],
      type: item.type ? item.type : '',
      accessService: item.accessService ? item.accessService : null
    }));
  } else {
    distributions = [];
  }
  return distributions;
};
