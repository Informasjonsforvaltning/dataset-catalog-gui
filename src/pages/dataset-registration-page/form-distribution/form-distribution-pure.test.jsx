import React from 'react';
import { shallow } from 'enzyme';
import {
  FormDistributionPure,
  renderDistributions,
  renderDistributionLandingpage
} from './form-distribution-pure';
import { distributionTypes } from './distribution-types';
import openlicenses from '../../../../test/fixtures/openlicenses';
import distribution from '../../../../test/fixtures/distributions';

let defaultProps;
let wrapper;

beforeEach(() => {
  const { openLicenseItems } = openlicenses;
  defaultProps = {
    initialValues: {
      distribution: distributionTypes(distribution),
      openLicenseItems
    }
  };
  wrapper = shallow(<FormDistributionPure {...defaultProps} />);
});

test('should render FormDistributionPure correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render renderDistributions correctly', () => {
  wrapper = shallow(renderDistributions(defaultProps));
  expect(wrapper).toMatchSnapshot();
});

test('should render renderDistributionLandingpage correctly', () => {
  wrapper = shallow(renderDistributionLandingpage(defaultProps));
  expect(wrapper).toMatchSnapshot();
});
