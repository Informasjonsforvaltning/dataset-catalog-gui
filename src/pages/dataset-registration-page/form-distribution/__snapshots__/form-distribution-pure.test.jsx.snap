// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormDistributionPure correctly 1`] = `
<form>
  <FieldArray
    component={[Function]}
    initialValues={
      Object {
        "distribution": Array [
          Object {
            "accessService": null,
            "accessURL": Array [
              "http://google.no",
            ],
            "conformsTo": Array [],
            "description": Object {
              "nb": "Beskrivelse av distribusjon",
            },
            "format": Array [
              "json",
              "xml",
            ],
            "id": "",
            "license": Object {
              "extraType": null,
              "prefLabel": null,
              "uri": "http://creativecommons.org/publicdomain/zero/1.0/",
            },
            "page": Array [
              Object {
                "extraType": null,
                "prefLabel": Object {
                  "nb": "",
                },
                "uri": "https://google.com",
              },
            ],
            "type": "API",
          },
        ],
        "openLicenseItems": Array [
          Object {
            "code": "CC BY 4.0",
            "prefLabel_no": "Creative Commons Navngivelse 4.0 Internasjonal",
            "uri": "http://creativecommons.org/licenses/by/4.0/",
          },
          Object {
            "code": "CC0 1.0",
            "prefLabel_no": "Creative Commons Universal Fristatus-erklÃ¦ring",
            "uri": "http://creativecommons.org/publicdomain/zero/1.0/",
          },
          Object {
            "code": "NLOD",
            "prefLabel_no": "Norsk lisens for offentlige data",
            "uri": "http://data.norge.no/nlod/",
          },
        ],
      }
    }
    isReadOnly={false}
    languages={Array []}
    name="distribution"
    onDeleteFieldAtIndex={[Function]}
    openLicenseItems={Array []}
  />
</form>
`;

exports[`should render renderDistributionLandingpage correctly 1`] = `<div />`;

exports[`should render renderDistributions correctly 1`] = `
<div>
  <button
    className="fdk-btn-no-border"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-plus mr-2"
    />
    Legg til distribusjon
  </button>
</div>
`;
